### ISSUE TITLE
[Bug]: Training loop never seeds PyTorch - results are not reproducible despite seed config

### ISSUE THREAD

**Issue by Adit-Jain-srm** [Adit-Jain-srm/NightmareNet#64]
## Problem Statement

The `configs/default.yaml` defines `seed: 42` and the README emphasizes reproducibility ("seed 42, Full methodology"). However, the training loop in `Trainer.__init__()` and `Trainer.train()` never calls:

- `torch.manual_seed(seed)`
- `torch.cuda.manual_seed_all(seed)`
- `numpy.random.seed(seed)` (for DataLoader worker seeding)
- `torch.backends.cudnn.deterministic = True`

Only the distortion generators and data generators seed Python's `random.seed()`. But PyTorch operations (weight initialization, dropout, DataLoader shuffling, gradient noise) remain non-deterministic.

This means:
- Running the same config twice produces different robustness scores
- The benchmark results in the paper draft are not reproducible by others
- The "reproducible benchmark methodology" claim in `docs/research/benchmark-v1.md` is false

**Evidence:**
```bash
rg "manual_seed|random.seed" nightmarenet/ --type py
# Only hits: data/generator.py and distortions/ (not trainer or pipeline)
```

**Relevant files:**
- `nightmarenet/training/trainer.py` (should seed in `__init__` or at start of `train()`)
- `configs/default.yaml` (defines `seed: 42` which is never consumed by training)

## Proposed Solution

Add a `_set_reproducibility(seed)` helper called at the start of `Trainer.train()`:

```python
def _set_reproducibility(seed: int) -> None:
    import random
    import numpy as np
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    if torch.cuda.is_available():
        torch.cuda.manual_seed_all(seed)
    torch.backends.cudnn.deterministic = True
    torch.backends.cudnn.benchmark = False
```

Also set `worker_init_fn` in DataLoaders to seed each worker process.

**Files affected:**
- `nightmarenet/training/trainer.py` - add seeding at start of train()
- Optionally: `nightmarenet/pipeline.py` - pass seed to DataLoader creation

## Acceptance Criteria

- [ ] `torch.manual_seed(config["seed"])` called before training starts
- [ ] `torch.cuda.manual_seed_all()` called for GPU reproducibility
- [ ] `numpy` and Python `random` seeded for DataLoader workers
- [ ] Two runs with identical config produce identical loss curves
- [ ] `cudnn.deterministic = True` when reproducibility is needed
- [ ] Test: two training runs with same seed produce same first-epoch loss

## Scope / Difficulty Estimate

Small. ~1-2 hours. Standard PyTorch reproducibility pattern. Well-documented in PyTorch docs.

**Comment by Sk-Uruj** [comment]
Hi @Adit-Jain-srm, I would love to fix this bug! 

I can update the training loop in `Trainer.__init__()` and `Trainer.train()` to correctly read the configuration seed and apply `torch.manual_seed`, `torch.cuda.manual_seed_all`, `numpy.random.seed`, and set the cuDNN deterministic flag to ensure the benchmark results are fully reproducible.

Could you please assign this issue to me? Thanks!

**Comment by Adit-Jain-srm** [comment]
Assigned, go ahead @Sk-Uruj
